force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -212,10 +260,21 @@ public with sharing class ClassesObjectsMethods {
      * @return The a positive number years between the two dates, or null if either date is null.
      */
     public static Integer calculateYearDifference(Date d1, Date d2) {
-
+        Integer diff;
+        if (d1 == null || d2 == null) {
+            return null;
+        } 
+        if (d1.year() > d2.year()){
+            diff = d1.year() - d2.year();
+        }
+        if (d2.year() > d1.year()) {
+            diff = d2.year() - d1.year();
+        }
+        if (d1.year() == d2.year()) {
+            diff = 0;
```

Comment:
```suggestion
 Integer diff= d2.year() - d1.year();

```
If you look at the example in the instruction it appears that d1 is being substracted from d2 only.